dnachisel/Location.py

```diff
@@ -42,18 +42,25 @@ def overlap_region(self, other_location):
         """Return the overlap span between two locations (None if None)."""
         strand = self.strand
         if other_location.start < self.start:
-            self, other_location = other_location, self
-
-        if other_location.start < self.end:
-            start = other_location.start
-            end = min(self.end, other_location.end)
-            strand = self.strand
-            return Location(start, end, strand)
+            left_location, right_location = other_location, self
         else:
+            left_location, right_location = self, other_location
+
+        if right_location.start >= left_location.end:
             return None
 
+        start = right_location.start
+        end = min(left_location.end, right_location.end)
+        strand = self.strand
```

Comment:
This is a duplicate line -- see line 43 -- just double checking if you had something else in mind.